streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -988,6 +988,18 @@ public void unregister(final Collection<TopicPartition> revokedChangelogs) {
             if (changelogMetadata != null) {
                 if (!changelogMetadata.state().equals(ChangelogState.REGISTERED)) {
                     revokedInitializedChangelogs.add(partition);
+
+                    // if the changelog is still in REGISTERED, it means it has not initialized and started
```

Comment:
> if the changelog is still in REGISTERED

We check `!changelogMetadata.state().equals(ChangelogState.REGISTERED)` above, so it seem we _are_ not in REGISTRED state here. Should the comment go somewhere else, or not use `if...` but say: `because...` ?